### PR TITLE
Implement basic repo model

### PR DESCRIPTION
**Pull request by mmalecki**

### CODE REVIEW

File: models/repo.go [hunk 1]
```diff
@@ -0,0 +1,57 @@
...
+    dberr, ok := err.(couchdb.DatabaseError)
+    if ok && dberr.StatusCode == 404 {
+      err = nil
+      ret = nil
+    }
+  }
+  return ret, err
+}
+
+func CreateRepo(repo *Repo) error {
```
Comment (yawnt) [inline models/repo.go:37]:
``` go
func (r *Repo) CreateRepo() error {

}
```

Comment (mmalecki) [inline models/repo.go:37]: Wouldn't it be `Create()` then tho?

Comment (Southern) [inline models/repo.go:37]: Yeah, just `Create()` if you're putting it on the struct.

**Comment by Southern** [comment]
Merging. :D

File: models/repo.go (hunk 1 above)
Comment (mmalecki) [inline models/repo.go:37]: So the API would be

``` go
repo := &Repo({
  ...
})

repo.Create()

foo := GetRepo("foo", "foo")
foo.Update(...)
```

?

**Pull request merged**